src/pyramid/security.py

```diff
@@ -363,3 +363,50 @@ def has_permission(self, permission, context=None):
             )  # should never happen
         principals = authn_policy.effective_principals(self)
         return authz_policy.permits(context, principals, permission)
+
+
+@implementer(ISecurityPolicy)
+class CompatibilitySecurityPolicy(object):
+    def __init__(self, authn_policy, authz_policy):
+        self.authn_policy = authn_policy
+        self.authz_policy = authz_policy
+
+    def identify(self, request):
+        return self.authn_policy.authenticated_userid(request)
+
+    def remember(self, request, userid, **kw):
+        return self.authn_policy.remember(request, userid, **kw)
+
+    def forget(self, request):
+        return self.authn_policy.forget(request)
+
+    def permits(self, context, request, identity, permission):
+        principles = self.authn_policy.effective_principles(request)
+        return self.authz_policy.permits(context, principles, permission)
+
+
+@implementer(ISecurityPolicy)
+class SessionSecurityPolicy(object):
+    def __init__(self, prefix='auth.'):
+        self.prefix = prefix or ''
+        self.identity_key = prefix + 'identity'
+
+    def identify(self, request):
+        return request.session[self.identity_key]
+
+    def remember(self, request, identity, **kw):
+        """ Store a userid in the session."""
+        request.session[self.identity_key] = identity
+        return []
+
+    def forget(self, request):
+        """ Remove the stored userid from the session."""
+        if self.identity_key in request.session:
+            del request.session[self.identity_key]
+        return []
+
+    def permits(self, context, request, identity, permission):
+        raise NotImplementedError(
+            '`SessionSecurityPolicy` does not provide an implementation for '
+            '`permits`.  You may wish to make your own subclass.'
+        )
```

Comment:
Instead of shipping `SessionSecurityPolicy` and other variants like `AuthTktSecurityPolicy` I think the goal should be to convert those policies to things closer to the `AuthTktCookieHelper` that can be used by people to write their own security policies.